CRM/Civicase/Hook/BuildForm/RestrictCaseEmailContacts.php

```diff
@@ -40,8 +40,8 @@ public function run(CRM_Core_Form $form, $formName) {
    *   True when the hook can run.
    */
   private function shouldRun() {
-    $isEmailForm = get_class($this->form) === CRM_Contact_Form_Task_Email::class;
-    $isCaseEmail = !empty($this->form->getVar('_caseId'));
+    $isEmailForm = get_class($this->form) === CRM_Case_Form_Task_Email::class;
+    $isCaseEmail = !empty(CRM_Utils_Request::retrieve('caseid', 'String'));
```

Comment:
I think given the form itself was changed, the `_caseId` is now probably either called `id` or `_id`, you can check if you can still get the case id using either:

`$this->form->getVar('_id')`

or

`$this->form->getVar('id')`

instead of `CRM_Utils_Request::retrieve('caseid', 'String')`

Reply:
given you told me on chat that neither getVar('_id') or getVar('id) are working, it is fine to keep it as is, my only note would be to change the 2nd parameter from `String` to  `Positive`.